client/txn.go

```diff
@@ -389,32 +383,37 @@ func (txn *Txn) send(calls ...proto.Call) error {
 		return nil
 	}
 
+	if txn.Proto.Status != proto.PENDING {
+		return util.Errorf("attempting to use %s transaction", txn.Proto.Status)
+	}
+
 	// haveTxnWrite tracks intention to write. This is in contrast to
 	// txn.Proto.Writing, which is set by the coordinator when the first
 	// intent has been created, and which lives for the life of the
 	// transaction.
 	var haveTxnWrite bool
```

Comment:
if you init `haveTxnWrite := proto.IsTransactionWrite(lastCall.Args)`, you can remove 409-410.